sync/options.go

```diff
@@ -21,18 +21,19 @@
 package sync
 
 import (
+	"runtime"
 	"time"
 
 	"github.com/m3db/m3x/instrument"
 )
 
 const (
-	defaultNumShards             = int64(2 ^ 4)
-	defaultKillWorkerProbability = 0.0001
+	defaultKillWorkerProbability = 0.001
```

Comment:
Do we need to edit the default grow kill probability in `config/pooling.go`?

Reply:
good call, done